languages/cpp/exercises/concept/basics/.meta/example.cpp

```diff
@@ -0,0 +1,15 @@
+#include "basics.h"
+
+int Lasagna::expectedMinutesInOven() { return 40; }
+
+int Lasagna::remainingMinutesInOven(int actualMinutesInOven) {
```

Comment:
This way of defining a class and its methods looks very different from that explained in the introduction. Should the example be similar to the expected solution or does it just need to be enough to make the tests pass in the CI? Unsure if there would be other impacts, like on the analyzer or other.

Reply:
Right, I have not thought much about this, I will add this in the introduction and talk about how usually C++ programs are divided in headers and sources.

Reply:
> Should the example be similar to the expected solution or does it just need to be enough to make the tests pass in the CI? Unsure if there would be other impacts, like on the analyzer or other.

In v2 the example implementation was just there to demonstrate that an implementation could pass the tests. With v3, the example is something more: an idiomatic implementation that could be referred to later. It should only use syntax that the student is familiar with based on the concepts in this exercise and its prerequisites. For more information, see [this description](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md#example-implementation-file).

Reply:
I agree. I believe this can be header-only

Reply:
We could argue that every Concept Exercise can be header only, they're all supposed to be pretty simple and small.

But I think we should show the student right away how things are usually done to avoid future confusion, maybe in the next exercises we can better explain why code is actually split between files.

Reply:
I agree with your argument, this can be introduced right away. However, we should add some indication how, for example,  `Lasagna::remainingMinutesInOven` is related to what is defined in the header file.